### DISCUSSION TITLE
Cancel button

### DISCUSSION THREAD

**Discussion by MichaelSchmidle** [formkit/formkit#947] [Ideas]
First of all, thanks for the great kit. Saves a ton of time!

One thing that I‘m missing is the ability to add a cancel button to form actions. Here‘s why:

I use a lot of forms in modals. Having a cancel button is the established way of closing such a form modal. Right now, it needs to be implemented separately which means that it doesn’t sit in the same spot as the submit button. Of course, one can implement a separate submit button next to the cancel button, too, and disable the built-in one. However, these buttons then sit on top of a form validation error, not below. Looks weird and is inconsistent design compared to using the built-in button elsewhere. (Not all forms sit in a modal and need a cancel button).

Such a cancel button would need to be capable of handling a @click prop to emit a closing event. And of course, it should have a different styling and customizable label. It could optionally also be used to clear the form.

What do you think?

**Comment by fenilli** [comment]
For your case I would think the `#actions` slot would be sufficient? about the sitting in the same spot this is mostly css so I would believe that a div with grid or flex will make it placed correctly.

**Reply by MichaelSchmidle** [reply]
Thanks, Gustavo, that approach works! I didn't really register the concept of slots or sections, as they're called in the docs.
